frontend-react/src/contexts/SessionContext.tsx

```diff
@@ -51,17 +60,33 @@ const SessionProvider = ({
         return activeMembership?.memberType === MemberType.PRIME_ADMIN;
     }, [activeMembership?.memberType]);
 
+    const updateSessionStartTime = (date: Date) => {
+        setState((prevState) => {
+            return { ...prevState, sessionStartTime: date };
+        });
+    };
+
+    const updateSessionTimeAggregate = (timeInSeconds: number) => {
+        setState((prevState) => {
+            return { ...prevState, sessionTimeAggregate: timeInSeconds };
+        });
+    };
+
+    const [state, setState] = useState<RSSessionContext>({
```

Comment:
Hmmm I don't really think this needs to exist in SessionContext.  Every time the state changes here (including the time updates) the reference to `state` will update, which can cause a full rerender of the entire SessionContext.

I think `visibilityState` stuff can just exist in App.tsx since that's the top-level component since `visibilityState` events only need to be bound in one place for a client-side app.  Or if we want to split it out, we can make it a separate hook 🤷

Reply:
Bumping this point 🐢

Reply:
My feeling is that we can continue to treat SessionContext as the place where we maintain auth state, but session _timing_ is conceptually separate from that -- like I don't think we should have to update all of the tests in this changeset to accommodate the updated SessionContext value 🤔 I think if we want to keep it encapsulated, we could break it out into a separate hook like `useSessionTimings` that'd 1) maintain the start time and session time sum, 2) return functions to start/pause/reset the session time, 3) bind to the document visibility state on render.

What are your thoughts about this?

Reply:
Chatted offline, but decided to remove all the work within `SessionContext` and just have a single instance of `sessionTime` exist within `App.tsx` so we don't have to share any variables across the app.